app/src/sandbox/scroll-spy/ScrollSpy.tsx

```diff
@@ -0,0 +1,80 @@
+import { MutableRefObject, ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+
+export interface IScrollSpyApi {
+    scrollToElement: (item?: string, selector?: string) => void;
+    currentActive: string;
+    setRef: (ref: HTMLElement) => void;
+}
+interface IScrollSpyProps {
+    root: HTMLElement;
+    elements?: Readonly<string[]>;
+    selector?: string;
+    initialActive?: string;
+    options?: IntersectionObserverInit;
+    children?: (api: IScrollSpyApi) => ReactNode;
+}
+
+export function useScrollSpy(
+    elements?: IScrollSpyProps['elements'],
+    initialActive?: IScrollSpyProps['initialActive'],
+    options?: IScrollSpyProps['options']
+) : IScrollSpyApi {
```

Comment:
You will use IScrollSpyProps interface for params and for ScrollSpyContainer you will extend it with IHasChildren interface